### PR TITLE
docs: additional application builder SSR conversion information

### PR DESCRIPTION
**Pull request by clydin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

N/A

## What is the new behavior?

This change updates the new build system migration guide with additional information regarding the steps needed to convert an existing SSR project to use the new integrated SSR capabilities.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: aio/content/guide/esbuild.md [hunk 1]
```diff
@@ -111,15 +111,31 @@ Many of the warnings will provide suggestions on how to remedy that problem.
 If it appears that a warning is incorrect or the solution is not apparent, please open an issue on [GitHub](https://github.com/angular/angular-cli/issues).
 Also, the later sections of this guide provide additional information on several specific cases as well as current known issues.
 
-For applications that are already using SSR, additional manual adjustments to code will be needed to update the SSR server code to support the new integrated SSR capabilities.
+For applications new to SSR, the [Angular SSR Guide](/guide/ssr) provides additional information regarding the setup process for adding SSR to an application.
+
+For applications that are already using SSR, additional manual adjustments will be needed to update the application SSR server to support the new integrated SSR capabilities.
+In the future, a schematic will make this migration process easier for existing applications.
```
Comment (dgp1130) [inline aio/content/guide/esbuild.md:117]: Nit: Looks like this line is duplicated at the end. Are you intending it to be in the both places? Maybe just the one at the end is fine?  I feel like this is too early in the doc for readers to understand what it means.

File: aio/content/guide/esbuild.md [hunk 2]
```diff
@@ -111,15 +111,31 @@ Many of the warnings will provide suggestions on how to remedy that problem.
 If it appears that a warning is incorrect or the solution is not apparent, please open an issue on [GitHub](https://github.com/angular/angular-cli/issues).
 Also, the later sections of this guide provide additional information on several specific cases as well as current known issues.
 
-For applications that are already using SSR, additional manual adjustments to code will be needed to update the SSR server code to support the new integrated SSR capabilities.
+For applications new to SSR, the [Angular SSR Guide](/guide/ssr) provides additional information regarding the setup process for adding SSR to an application.
+
+For applications that are already using SSR, additional manual adjustments will be needed to update the application SSR server to support the new integrated SSR capabilities.
```
Comment (dgp1130) [inline aio/content/guide/esbuild.md:116]: Readability: "update the application ~SSR~ server". Just reads a little better and we've said "SSR" enough in this sentence.

File: aio/content/guide/esbuild.md [hunk 3]
```diff
@@ -111,15 +111,31 @@ Many of the warnings will provide suggestions on how to remedy that problem.
...
 
 * `app-shell`
 * `prerender`
 * `server`
 * `ssr-dev-server`
 
-The [Angular SSR Guide](/guide/ssr) provides additional information regarding the new setup process for SSR.
+The `ng update` process will automatically remove usages of the `@nguniversal` scope packages where some of these builders were previously located.
+The new `@angular/ssr` package will also be automatically added and used with configuration and code being adjusted during the update.
+To convert from the separate SSR builders to the integrated capabilities of the `application` builder, there are several required manual steps.
```
Comment (dgp1130) [inline aio/content/guide/esbuild.md:127]: Consider: Should we point out that staying on the old builders from `@angular/ssr` is fine for using v17? I want to make sure it's clear that migrating to v17 does _not_ require going through all these steps. This part is just for the new build improvements.

File: aio/content/guide/esbuild.md [hunk 4]
```diff
@@ -111,15 +111,31 @@ Many of the warnings will provide suggestions on how to remedy that problem.
...
+However, as each application is different, there may be more application specific changes needed beyond these to complete the process.
+
+1. Combine SSR builder options into the `application` builder options in the `angular.json` file.
+2. Combine server TypeScript configuration from `tsconfig.server.json` into `tsconfig.app.json`.
+3. Remove and/or update any `npm` scripts referencing the old builder targets.
+4. Update application server code to remove Webpack specific elements.
+5. Remove any CommonJS assumptions in the application server code.
```
Comment (dgp1130) [inline aio/content/guide/esbuild.md:134]: Suggestion: I think a little more detail for the motivation or goal of each point would be helpful in communicating the objective for each change. Maybe something along the lines of:

> 1. Combine SSR builder options into a new `application` builder in the `angular.json` file. Remove `browser`, `server`, `ssr-dev-server`, and `prerender` builders. `application` includes equivalent functionality for all of them.
> 2. Move server TypeScript configuration from `tsconfig.server.json` into `tsconfig.app.json` and delete `tsconfig.server.json`. `application` builder performs a single type-check pass for both the client and server builds, using a single configuration.
> 3. Remove and/or update any `npm` scripts referencing deleted builder targets. `ng build` with `application` builder will automatically run prerendering and generate a Node server.
> 4. Update application server code to remove Webpack-specific elements.
> 5. Remove any [CommonJS](https://nodejs.org/api/modules.html#modules-commonjs-modules) assumptions in the application server code such as `require`, `__dirname`, or `__filename`. `application` builder emits native [ES modules](https://nodejs.org/api/esm.html).

Feel free to adjust, not trying to rewrite things entirely. Just trying to get across the idea of putting enough context for readers to understand the objective behind each change.

Comment (dgp1130) [inline aio/content/guide/esbuild.md:134]: Question: One thing I'm honestly a little unclear on is how this affects dependencies. Can dependencies still use CommonJS and will those be converted to ESM? Or do we need to use native ESM-compatible dependencies as well?

File: aio/content/guide/esbuild.md [hunk 5]
```diff
@@ -111,15 +111,31 @@ Many of the warnings will provide suggestions on how to remedy that problem.
...
+
+To provide an example of the specific changes needed to convert an existing v16 project, a repository with individual commits outlining the steps is available:
```
Comment (dgp1130) [inline aio/content/guide/esbuild.md:136]: Active voice: "This repository demonstrates an example of the specific changes needed to upgrade an existing v16 SSR application to Angular v17 and then migrate to the new `application` builder."

**Review by dgp1130 — COMMENTED** [review]
Thanks for putting this together, definitely a big improvement!

File: aio/content/guide/esbuild.md (hunk 5 above)
Comment (clydin) [reply aio/content/guide/esbuild.md]: Removed this completely for now so we can get these updates merged.
Once the example is ready we can add back the line.

File: aio/content/guide/esbuild.md (hunk 4 above)
Comment (clydin) [reply aio/content/guide/esbuild.md]: It doesn't.  Bundling handles commonjs dependencies.  Only application code needs to be ESM compliant.
Added a sentence about this.

Comment (clydin) [reply aio/content/guide/esbuild.md]: Updated.

File: aio/content/guide/esbuild.md (hunk 3 above)
Comment (clydin) [reply aio/content/guide/esbuild.md:127]: This is in the section about how to use the new build improvements (specifically the `application` builder).
But added a line about how all applications regardless of builder will have that change.

File: aio/content/guide/esbuild.md [hunk 6]
```diff
@@ -111,15 +111,35 @@ Many of the warnings will provide suggestions on how to remedy that problem.
...
 * `ssr-dev-server`
 
-The [Angular SSR Guide](/guide/ssr) provides additional information regarding the new setup process for SSR.
+The `ng update` process will automatically remove usages of the `@nguniversal` scope packages where some of these builders were previously located.
+The new `@angular/ssr` package will also be automatically added and used with configuration and code being adjusted during the update.
+All updated applications, even those that stay on the `browser` builder, will have this change from the v17 update.
+To convert from the separate SSR builders to the integrated capabilities of the `application` builder, there are several required manual steps.
+However, as each application is different, there may be more application specific changes needed beyond these to complete the process.
+
+1. Combine the options for the above mentioned SSR builders into the `application` builder options within the `angular.json` file.
```
Comment (alan-agius4) [inline aio/content/guide/esbuild.md:130]: There are some missing steps here. You can take a look at https://github.com/alan-agius4/angular-cli-use-application-builder/commit/1defdb93a7f508662bc427439e51505668bf84cd

As discussed, for the **interim** we should add a link to that here.

File: aio/content/guide/esbuild.md (hunk 3 above)
Comment (dgp1130) [reply aio/content/guide/esbuild.md:127]: True, but I do think it's relevant here and worth including.

> All updated applications, even those that stay on the `browser` builder, will have this change from the v17 update.

I'm not sure I'm following "will have this change". I don't think I would understand that reading this blind. Maybe something like:

> `@angular/ssr` supports `browser` builder as well as `browser-esbuild` and `application` builder.

I'm not super happy with that either, but I'll defer to you on the specifics. However you want to handle it.

**Review by dgp1130 — APPROVED** [review]

**Comment by alxhub** [comment]
This PR was merged into the repository by commit f615f4f90d3a053de2f273f6bd6d3e00ebfbb790.

File: aio/content/guide/esbuild.md (hunk 3 above)
Comment (alan-agius4) [reply aio/content/guide/esbuild.md:127]: browser-builder is not supported by @angular/ssr

Comment (clydin) [reply aio/content/guide/esbuild.md:127]: yeh. i took that part out.

**Pull request closed without merging**